docs/integrations/code-conventions.md

```diff
@@ -885,6 +885,42 @@ return_results(results)
 ```
 **Note:** By default, ignore_auto_extract is set to ```False```.
 
+### Pagination in integration commands
+When working on a command that supports pagination (usually has API parameters like`page` and/or `page size`) with a maximal page size enforced by the API, our best practice is to create a command to support two different use-cases with these 3 corresponding integer arguments:
+1. `page` 
+2. `page_size` 
+3. `limit` 
+
+**Manual Pagination:** The user wants to control the pagination on its own by using the `page` and `page size` arguments. To achieve this, the command will simply pass the `page` and `page size` values on to the API request.
+
+**Automatic Pagination:** The user does not want to work with pages only with a number of results argument. In this case, the `limit` argument will be used to aggregate results by iterating over the necessary pages from the first page until collecting all the needed results. This implies a pagination loop mechanism will be implemented behind the scenes. For example, if the limit value received is 250 and the maximal page size enforced by the API is 100, the command will need to perform 3 API calls (pages 1,2, and 3) to collect the 250 requested results.
+
+### Credentials
+When working on integrations that require user credentials (such as username/password, API token/key, etc..) the best practice is to use the `credentials` parameter type.
+Example when using username and password:
+In the YML file:
+```yml
+- display: Username
+  name: credentials
+  type: 9
+  required: true
+```
+In Demisto UI:
+<img width="758" src="../doc_imgs/integrations/credentials_username_password.png"></img>
+
+Example when using an API Token/Key:
+```yml
+- displaypassword: API Token
+  name: credentials
+  type: 9
+  required: false
+  hiddenusername: true
+```
+In Demisto UI:
+<img width="758" src="../doc_imgs/integrations/credentials_api_token.png"></img>
+  
+Using credentials parameter type is always recommended (even when working with API token\key) as it provides the user the flexibility of using the [XSOAR credentials vault](https://xsoar.pan.dev/docs/reference/articles/managing-credentials) feature when configuring the integration for the first time.
```

Comment:
**Using username and password:**

- **In Demisto UI:**
![image](../doc_imgs/integrations/credentials_username_password.png)

- **In the YML file:**
```yml
- display: Username
  name: credentials
  type: 9
  required: true
```

**Using an API Token/Key:**

- **In Demisto UI:**
![image](../doc_imgs/integrations/credentials_api_token.png)

- **In the YML file:**
```yml
- displaypassword: API Token
  name: credentials
  type: 9
  required: false
  hiddenusername: true
```
  
Using credentials parameter type is always recommended (even when working with API token\key) as it provides the user the flexibility of using the [XSOAR credentials vault](../docs/reference/articles/managing-credentials) feature when configuring the integration for the first time.